src/building/data_transfer.c

```diff
@@ -39,6 +40,7 @@ int building_data_transfer_copy(building *b)
     } else {
         memset(&data, 0, sizeof(data));
         data.data_type = data_type;
+        data.mothball_state = b->state;
```

Comment:
I don't think it's a good idea to copy the building state.

Maybe a better way to do this is to add :

`data.building_is_mothballed = b->state == BUILDING_STATE_MOTHBALLED ? 1 : 0`